lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/LeastSquaresExpansion.cxx

```diff
@@ -0,0 +1,230 @@
+  const UnsignedInteger sampleSize = inputSample_.getSize();
+  for (UnsignedInteger j = 0; j < outputDimension; ++j)
+    {
+      const Sample marginalOutputSample(outputSample_.getMarginal(j));
+      const Point rhs(marginalOutputSample.asPoint());
+      const Point coeffsJ(leastSquaresMethod.solve(rhs));
+      for (UnsignedInteger i = 0; i < basisSize_; ++i)
+        coefficients(i, j) = coeffsJ[i];
+      // Now the two errors
+      const Scalar quadraticResidual = (weightedDesign * coeffsJ - rhs).normSquare();
+      residuals[j] = std::sqrt(quadraticResidual) / sampleSize;
+      const Scalar empiricalError = quadraticResidual / sampleSize;
+      relativeErrors[j] = empiricalError / marginalOutputSample.computeVariance()[0];
+    }
+  // Build the result
+  result_ = FunctionalChaosResult(inputSample_, outputSample_, distribution_, transformation_, inverseTransformation_, basis_, activesFunctions_, coefficients, designProxy_.getBasis(), residuals, relativeErrors);
+}
+
+/* Method to get/set the active functions */
+Indices LeastSquaresExpansion::getActivesFunctions() const
+{
+  return activesFunctions_;
+}
+
+void LeastSquaresExpansion::setActivesFunctions(const Indices & activesFunctions)
+{
+  if (!activesFunctions.check(basisSize_)) throw InvalidArgumentException(HERE) << "Error: the active functions must have indices less than " << basisSize_;
+  activesFunctions_ = activesFunctions;
+}
+
+
+/* String converter */
+String LeastSquaresExpansion::__repr__() const
+{
+  return OSS() << "class=" << getClassName()
+         << " basis_=" << basis_
+         << " basisSize_=" << basisSize_
+         << " activesFunctions_=" << activesFunctions_
+         << " designProxy_=" << designProxy_
+         << " transformation_=" << transformation_
+         << " inverseTransformation_=" << inverseTransformation_;
+}
+
+
+/* String converter */
+String LeastSquaresExpansion::__str__(const String & ) const
+{
+  return __repr__();
+}
+
+/* Method save() stores the object through the StorageManager */
+void LeastSquaresExpansion::save(Advocate & adv) const
+{
+  FunctionalChaosAlgorithm::save(adv);
+  adv.saveAttribute( "basis_", basis_ );
+  adv.saveAttribute( "basisSize_", basisSize_ );
+  adv.saveAttribute( "activesFunctions_", activesFunctions_ );
+  adv.saveAttribute( "methodName_", methodName_ );
+  adv.saveAttribute( "designProxy_", designProxy_ );
+  adv.saveAttribute( "transformation_", transformation_ );
+  adv.saveAttribute( "inverseTransformation_", inverseTransformation_ );
+}
+
+
+/* Method load() reloads the object from the StorageManager */
+void LeastSquaresExpansion::load(Advocate & adv)
+{
+  FunctionalChaosAlgorithm::load(adv);
+  adv.loadAttribute( "basis_", basis_ );
+  adv.loadAttribute( "basisSize_", basisSize_ );
+  adv.loadAttribute( "activesFunctions_", activesFunctions_ );
+  adv.loadAttribute( "methodName_", methodName_ );
+  adv.loadAttribute( "designProxy_", designProxy_ );
```

Comment:
why do we want to save the designproxy ? it can be big
in practice we only save the result, so I guess it doesnt matter, but I dont recall your argument last time